### PR TITLE
CTAD for simple_ptr_holder

### PR DESCRIPTION
**Pull request by havogt**
Not sure if it's worth trading aggregate for CTAD. When I looked at make_simple_ptr_holder, I missed that aggregate CTAD is a C++20 feature.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: include/gridtools/sid/simple_ptr_holder.hpp [hunk 1]
```diff
@@ -30,22 +30,31 @@ namespace gridtools {
             template <class T>
             struct simple_ptr_holder {
                 T m_val;
+
+#if defined(__cpp_deduction_guides) and __cpp_deduction_guides >= 201907
+// CTAD for aggregates works
+#else
+                // Enables CTAD in C++17.
+                simple_ptr_holder() = default;
```
Comment (havogt) [inline include/gridtools/sid/simple_ptr_holder.hpp:38]: default ctor only there for some test cases that rely on default init.

File: include/gridtools/sid/simple_ptr_holder.hpp [hunk 2]
```diff
@@ -30,22 +30,31 @@ namespace gridtools {
...
+                GT_TARGET GT_FORCE_INLINE constexpr simple_ptr_holder(T const &ptr) : m_val{ptr} {}
+#endif
                 GT_TARGET GT_FORCE_INLINE constexpr T const &operator()() const { return m_val; }
             };
 
             template <class T>
-            constexpr simple_ptr_holder<T> make_simple_ptr_holder(T const &ptr) {
+            [[deprecated("use simple_ptr_holder class template argument deduction")]] constexpr simple_ptr_holder<T>
```
Comment (anstaf) [inline include/gridtools/sid/simple_ptr_holder.hpp:45]: Are you sure you want to keep the maker? I thought it is not the part of the public API.

**Review by anstaf — APPROVED** [review]

File: include/gridtools/sid/simple_ptr_holder.hpp (hunk 2 above)
Comment (havogt) [reply include/gridtools/sid/simple_ptr_holder.hpp:45]: I'll keep it for now.

**Pull request merged**